<!DOCTYPE html>

<html lang="en" data-content_root="./">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>catcheR_load - Data loading &#8212; catcheR 2.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=d1102ebc" />
    <link rel="stylesheet" type="text/css" href="_static/basic.css?v=686e5160" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css?v=27fed22d" />
    <script src="_static/documentation_options.js?v=60dbed4a"></script>
    <script src="_static/doctools.js?v=9bcbadda"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="catcheR_pseudotime - Pseudotime" href="catcheR_pseudotime.html" />
    <link rel="prev" title="catcheR_sortcatch - Barcode reassignment" href="catcheR_sortcatch.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="catcher-load-data-loading">
<h1>catcheR_load - Data loading<a class="headerlink" href="#catcher-load-data-loading" title="Link to this heading">¶</a></h1>
<p>This step enables the loading of single-cell data generated by either <code class="docutils literal notranslate"><span class="pre">catcheR_10Xcatch</span></code> or <code class="docutils literal notranslate"><span class="pre">catcheR_scicatch</span></code>, following GTF-based annotation. The data is imported into a <strong>Monocle</strong> object, where experimental design information is added, followed by normalization and clustering.</p>
<section id="preparation">
<h2>Preparation<a class="headerlink" href="#preparation" title="Link to this heading">¶</a></h2>
<p>Before running <code class="docutils literal notranslate"><span class="pre">catcheR_load</span></code>, prepare the following in a new working folder:</p>
<ol class="arabic simple">
<li><p>Copy the count matrix annotated with gene names from the previous step
(e.g., <code class="docutils literal notranslate"><span class="pre">filtered_annotated_silencing_matrix_complete_all_samples.csv</span></code>)</p></li>
<li><p>Copy the file <code class="docutils literal notranslate"><span class="pre">rc_barcodes_genes.csv</span></code></p></li>
<li><p>Create a newline-separated plain text file listing the <strong>control genes</strong>
(e.g., SCR, B2M)</p></li>
<li><p>Create a newline-separated plain text file listing the <strong>control samples</strong> (if any)
(e.g., 1, 3). These sample names should match those used by <code class="docutils literal notranslate"><span class="pre">aggr</span></code>
(see the input CSV file used for <code class="docutils literal notranslate"><span class="pre">aggr</span></code>)</p></li>
<li><p>Create a newline-separated plain text file listing the <strong>sample replicate labels</strong>
(e.g. batch1, batch1, batch2, batch2).
The order must match the sample order in the input matrix exactly.
This file is required for downstream <strong>batch-aware analyses</strong>.
If your dataset includes multiple experiments or processing batches, batch correction is recommended.</p></li>
<li><p>Create a <strong>CSV file</strong> listing each sample along with its <strong>annotation name</strong> (required).
This will be used in plots instead of the sample number.
Example file available on <a class="reference external" href="https://github.com/alessandro-bertero/catcheR/blob/dev/input_examples/samples.csv">GitHub</a></p></li>
<li><p><em>(Optional)</em> Create a newline-separated plain text file listing <strong>genes of interest</strong>
whose expression will be visualized on the UMAP.</p></li>
</ol>
</section>
<section id="running-catcher-load">
<h2>Running <code class="docutils literal notranslate"><span class="pre">catcheR_load</span></code><a class="headerlink" href="#running-catcher-load" title="Link to this heading">¶</a></h2>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">catcheR_load</span><span class="p">(</span>
<span class="w">  </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;docker&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">folder</span><span class="p">,</span>
<span class="w">  </span><span class="n">expression.matrix</span><span class="p">,</span>
<span class="w">  </span><span class="n">control_genes</span><span class="p">,</span>
<span class="w">  </span><span class="n">control_samples</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span>
<span class="w">  </span><span class="n">replicates</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span>
<span class="w">  </span><span class="n">sample_names</span><span class="p">,</span>
<span class="w">  </span><span class="n">resolution</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8e-4</span><span class="p">,</span>
<span class="w">  </span><span class="n">genes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span>
<span class="p">)</span>
</pre></div>
</div>
<p><strong>Example usage:</strong></p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">catcheR_load</span><span class="p">(</span>
<span class="w">  </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;docker&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;/path/to/working/folder/&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">expression.matrix</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;annotated_silencing_matrix_complete_all_samples.csv&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">control_genes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;controls.txt&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">control_samples</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;noTET.txt&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">replicates</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;replicates.txt&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">sample_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;samples.csv&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">resolution</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8e-4</span><span class="p">,</span>
<span class="w">  </span><span class="n">genes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;genelist.txt&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">resolution</span></code> argument sets the resolution parameter used by Monocle’s <code class="docutils literal notranslate"><span class="pre">cluster_cells</span></code> function.</p>
</section>
<section id="outputs">
<h2>Outputs<a class="headerlink" href="#outputs" title="Link to this heading">¶</a></h2>
<p>Running <code class="docutils literal notranslate"><span class="pre">catcheR_load</span></code> produces the following outputs:</p>
<ol class="arabic">
<li><p><code class="docutils literal notranslate"><span class="pre">expression_data.csv</span></code> and <code class="docutils literal notranslate"><span class="pre">cell_metadata.csv</span></code>
These can be used to create a Monocle Cell Data Set (CDS) and are also bundled in <code class="docutils literal notranslate"><span class="pre">starting_cds.RData</span></code>, the ready-to-load R object.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">UMAP.pdf</span></code>
Dimensionality reduction UMAP plot.</p>
<img alt="_images/UMAP.pdf" src="_images/UMAP.pdf" />
</li>
<li><p><code class="docutils literal notranslate"><span class="pre">UMAP_gene_expression.pdf</span></code>
Gene expression overlay on UMAP using genes from the <code class="docutils literal notranslate"><span class="pre">genes</span></code> argument.</p>
<img alt="_images/UMAP_gene_expression.pdf" src="_images/UMAP_gene_expression.pdf" />
</li>
<li><p><code class="docutils literal notranslate"><span class="pre">UMAP_clustering.pdf</span></code>
Clustering result visualized on UMAP at the specified resolution.</p>
<img alt="_images/UMAP_clustering.pdf" src="_images/UMAP_clustering.pdf" />
</li>
<li><p><code class="docutils literal notranslate"><span class="pre">processed_cds.RData</span></code>
The Monocle CDS after normalization, dimensionality reduction, clustering, and trajectory inference.</p></li>
</ol>
</section>
<section id="compatibility">
<h2>Compatibility<a class="headerlink" href="#compatibility" title="Link to this heading">¶</a></h2>
<p>At the end of this step, your data will be structured for use with the <code class="docutils literal notranslate"><span class="pre">monocle3</span></code> package.</p>
<p>However, it is also possible to switch to other frameworks such as <strong>Seurat</strong> or <strong>Scanpy</strong>:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">SeuratWrappers</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">Seurat</span><span class="p">)</span>
<span class="n">seurat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.Seurat</span><span class="p">(</span><span class="n">cds</span><span class="p">,</span><span class="w"> </span><span class="n">assay</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">)</span>
<span class="n">scanpy_sce</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.SingleCellExperiment</span><span class="p">(</span><span class="n">seurat</span><span class="p">)</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>For standard <strong>iPS2-seq</strong> perturbation analysis, always continue using the <code class="docutils literal notranslate"><span class="pre">CDS</span></code> object generated by <code class="docutils literal notranslate"><span class="pre">catcheR</span></code>.</p>
</div>
</section>
<section id="next-steps">
<h2>Next steps<a class="headerlink" href="#next-steps" title="Link to this heading">¶</a></h2>
<p>The following analyses can be performed after this step:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">catcheR_pseudotime</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">catcheR_modules</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">catcheR_enrichment</span></code></p></li>
</ul>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">catcheR</a></h1>









<search id="searchbox" style="display: none" role="search">
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" placeholder="Search"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script><h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_design.html">catcheR_design - Oligonucleotide Design</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_step1QC.html">catcheR_step1QC - Pooled Cloning Step 1 QC</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_step2QC.html">catcheR_step2QC - Pooled Cloning Step 2 and hiPSC Genome Editing QC</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_scicounts.html">catcheR_scicounts - Obtain iPS2-sci-seq Count Matrices</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_10Xcatch.html">catcheR_10Xcatch - iPS2-10X-seq Perturbation Deconvolution</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_10XcatchQC.html">catcheR_10XcatchQC - Fine-Tune iPS2-10X-seq Perturbation Assignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_filtercatch.html">catcheR_filtercatch - Fine-Tune iPS2-10X-seq Perturbation Assignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_10Xnocatch.html">catcheR_10Xnocatch - Identify cells expressing no shRNA</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_merge.html">catcheR_merge - Merge multiple samples</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_scicatch.html">caycheR_scicatch - iPS2-sci-seq perturbation deconvolution</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_scicatchQC.html">catcheR_scicatchQC - Fine-tune iPS2-sci-seq perturbation assignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_scinocatch.html">catcheR_scinocatch - Identify cells expressing no shRNA in iPS2-sci-seq</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_sortcatch.html">catcheR_sortcatch - Barcode reassignment</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">catcheR_load - Data loading</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#preparation">Preparation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-catcher-load">Running <code class="docutils literal notranslate"><span class="pre">catcheR_load</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#outputs">Outputs</a></li>
<li class="toctree-l2"><a class="reference internal" href="#compatibility">Compatibility</a></li>
<li class="toctree-l2"><a class="reference internal" href="#next-steps">Next steps</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_pseudotime.html">catcheR_pseudotime - Pseudotime</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_modules.html">catcheR_modules - Gene Modules</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_enrichment.html">catcheR_enrichment - Enrichment / Depletion Analysis</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="catcheR_sortcatch.html" title="previous chapter">catcheR_sortcatch - Barcode reassignment</a></li>
      <li>Next: <a href="catcheR_pseudotime.html" title="next chapter">catcheR_pseudotime - Pseudotime</a></li>
  </ul></li>
</ul>
</div>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2025, Maria Luisa Ratto.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 8.1.3</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 1.0.0</a>
      
      |
      <a href="_sources/catcheR_load.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>