<!DOCTYPE html>

<html lang="en" data-content_root="./">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>catcheR_sortcatch - Barcode reassignment &#8212; catcheR 2.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=d1102ebc" />
    <link rel="stylesheet" type="text/css" href="_static/basic.css?v=686e5160" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css?v=27fed22d" />
    <script src="_static/documentation_options.js?v=60dbed4a"></script>
    <script src="_static/doctools.js?v=9bcbadda"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="catcheR_load - Data loading" href="catcheR_load.html" />
    <link rel="prev" title="catcheR_scinocatch - Identify cells expressing no shRNA in iPS2-sci-seq" href="catcheR_scinocatch.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="catcher-sortcatch-barcode-reassignment">
<h1>catcheR_sortcatch - Barcode reassignment<a class="headerlink" href="#catcher-sortcatch-barcode-reassignment" title="Link to this heading">¶</a></h1>
<p><code class="docutils literal notranslate"><span class="pre">catcheR_sortcatch</span></code> is an optional function that corrects the annotated cell-by-gene count matrix obtained with <code class="docutils literal notranslate"><span class="pre">catcheR_10Xcatch</span></code> or <code class="docutils literal notranslate"><span class="pre">catcheR_scicatch</span></code>. It reassigns the perturbation of any cell belonging to an hPSC clone with reliable evidence of a shRNA-barcode swap, based on the results of <code class="docutils literal notranslate"><span class="pre">catcheR_step1QC</span></code>.</p>
<section id="steps">
<h2>Steps<a class="headerlink" href="#steps" title="Link to this heading">¶</a></h2>
<ol class="arabic">
<li><p>In a new working folder:</p>
<ol class="loweralpha simple">
<li><p>Copy the annotated count matrix (e.g., <code class="docutils literal notranslate"><span class="pre">silencing_matrix.csv</span></code>)</p></li>
<li><p>Copy the CSV file with the list of UCI-BCs with reliable evidence of a shRNA-barcode swap (e.g., <code class="docutils literal notranslate"><span class="pre">reliable_clones_swaps.csv</span></code>; output of step <span class="xref std std-ref">step-SPFourStepEleven</span> from <span class="xref std std-ref">subsec-SPFourThree</span>)
.. _step-SPFourStepTwelve:</p></li>
</ol>
</li>
<li><p>Run <code class="docutils literal notranslate"><span class="pre">catcheR_sortcatch</span></code>:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">catcheR_sortcatch</span><span class="p">(</span>
<span class="w">    </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;sudo&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;docker&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="n">folder</span><span class="p">,</span>
<span class="w">    </span><span class="n">expression.matrix</span><span class="p">,</span>
<span class="w">    </span><span class="n">swaps</span>
<span class="p">)</span>
</pre></div>
</div>
<p><strong>Example usage:</strong></p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">catcheR_sortcatch</span><span class="p">(</span>
<span class="w">    </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;docker&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;path/to/working/folder&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">expression.matrix</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;silencing_matrix.csv&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">swaps</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;reliable_clones_swaps.csv&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</li>
</ol>
</section>
<section id="arguments">
<h2>Arguments<a class="headerlink" href="#arguments" title="Link to this heading">¶</a></h2>
<ul class="simple">
<li><p><strong>group</strong>: string with two options: <code class="docutils literal notranslate"><span class="pre">sudo</span></code> or <code class="docutils literal notranslate"><span class="pre">docker</span></code>, depending on the user group (<a class="reference external" href="https://docs.docker.com/engine/install/linux-postinstall/">info</a>)</p></li>
<li><p><strong>folder</strong>: string with the working folder path</p></li>
<li><p><strong>expression.matrix</strong>: string with the filename of the annotated count matrix CSV</p></li>
<li><p><strong>swaps</strong>: string with the filename of the swap list CSV file (see step <span class="xref std std-ref">step-SPFourStepTwelve</span>)</p></li>
</ul>
</section>
<section id="output">
<h2>Output<a class="headerlink" href="#output" title="Link to this heading">¶</a></h2>
<p><code class="docutils literal notranslate"><span class="pre">catcheR_sortcatch</span></code> outputs an updated annotated gene expression matrix CSV file called <code class="docutils literal notranslate"><span class="pre">silencing_matrix_updated.csv</span></code>, in which the “BC” and “GENE” annotations have been corrected to reflect the actual shRNA encoded in each cell.</p>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">catcheR</a></h1>









<search id="searchbox" style="display: none" role="search">
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" placeholder="Search"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script><h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="gettingstarted.html#notes-on-gene-annotation">Notes on gene annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_design.html">catcheR_design - Oligonucleotide Design</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_step1QC.html">catcheR_step1QC - Pooled Cloning Step 1 QC</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_step2QC.html">catcheR_step2QC - Pooled Cloning Step 2 and hiPSC Genome Editing QC</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_scicounts.html">catcheR_scicounts - Obtain iPS2-sci-seq Count Matrices</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_10Xcatch.html">catcheR_10Xcatch - iPS2-10X-seq Perturbation Deconvolution</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_10XcatchQC.html">catcheR_10Xcatch - Fine-Tune iPS2-10X-seq Perturbation Assignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_filtercatch.html">catcheR_filtercatch - Fine-Tune iPS2-10X-seq Perturbation Assignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_10Xnocatch.html">catcheR_10Xnocatch - Identify cells expressing no shRNA</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_merge.html">catcheR_merge - Merge multiple samples</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_scicatch.html">caycheR_scicatch - iPS2-sci-seq perturbation deconvolution</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_scicatchQC.html">catcheR_scicatchQC - Fine-tune iPS2-sci-seq perturbation assignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_scinocatch.html">catcheR_scinocatch - Identify cells expressing no shRNA in iPS2-sci-seq</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">catcheR_sortcatch - Barcode reassignment</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#steps">Steps</a></li>
<li class="toctree-l2"><a class="reference internal" href="#arguments">Arguments</a></li>
<li class="toctree-l2"><a class="reference internal" href="#output">Output</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_load.html">catcheR_load - Data loading</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_pseudotime.html">catcheR_pseudotime - Pseudotime</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_modules.html">catcheR_modules - Gene Modules</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_enrichment.html">catcheR_enrichment - Enrichment / Depletion Analysis</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="catcheR_scinocatch.html" title="previous chapter">catcheR_scinocatch - Identify cells expressing no shRNA in iPS2-sci-seq</a></li>
      <li>Next: <a href="catcheR_load.html" title="next chapter">catcheR_load - Data loading</a></li>
  </ul></li>
</ul>
</div>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2025, Maria Luisa Ratto.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 8.1.3</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 1.0.0</a>
      
      |
      <a href="_sources/catcheR_sortcatch.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>