<!DOCTYPE html>

<html lang="en" data-content_root="./">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>catcheR_filtercatch - Fine-Tune iPS2-10X-seq Perturbation Assignment &#8212; catcheR 2.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=d1102ebc" />
    <link rel="stylesheet" type="text/css" href="_static/basic.css?v=686e5160" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css?v=27fed22d" />
    <script src="_static/documentation_options.js?v=60dbed4a"></script>
    <script src="_static/doctools.js?v=9bcbadda"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="catcheR_10Xnocatch - Identify cells expressing no shRNA" href="catcheR_10Xnocatch.html" />
    <link rel="prev" title="catcheR_10Xcatch - Fine-Tune iPS2-10X-seq Perturbation Assignment" href="catcheR_10XcatchQC.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="catcher-filtercatch-fine-tune-ips2-10x-seq-perturbation-assignment">
<h1>catcheR_filtercatch - Fine-Tune iPS2-10X-seq Perturbation Assignment<a class="headerlink" href="#catcher-filtercatch-fine-tune-ips2-10x-seq-perturbation-assignment" title="Link to this heading">¶</a></h1>
<p>Still within the same working directory used for <code class="xref py py-func docutils literal notranslate"><span class="pre">catcheR_10Xcatch()</span></code>, you can re-apply filtering with user-defined thresholds by running:</p>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="nf">catcheR_filtercatch</span><span class="p">(</span>
<span class="w">    </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;docker&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sudo&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="n">folder</span><span class="p">,</span>
<span class="w">    </span><span class="n">expression.matrix</span><span class="p">,</span>
<span class="w">    </span><span class="n">UMI.count</span><span class="p">,</span>
<span class="w">    </span><span class="n">percentage</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span>
<span class="w">    </span><span class="n">ratio</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">,</span>
<span class="w">    </span><span class="n">sample</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span>
<span class="p">)</span>
</pre></div>
</div>
<p><strong>Arguments</strong>:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">group</span></code>: either <cite>“docker”</cite> or <cite>“sudo”</cite>, depending on your system permissions</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">folder</span></code>: string with the working directory path</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">expression.matrix</span></code>: string with the filename of the cell-by-gene count matrix (CSV format)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">UMI.count</span></code>: <em>integer</em> threshold for the minimum number of UMIs per UCI to be considered valid</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">percentage</span></code>: <em>integer</em> percentage threshold for a UCI to be accepted in a cell
(e.g., if <code class="docutils literal notranslate"><span class="pre">percentage</span> <span class="pre">=</span> <span class="pre">15</span></code>, the UCI must account for ≥15% of all UMI in that cell)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">ratio</span></code>: <em>integer</em> minimum ratio of the top UMI count to the second-highest in a cell to accept the top UCI (default = 5)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sample</span></code>: <em>integer</em> specifying the sample number (default = 1). Run this step separately for each sample.</p></li>
</ul>
<p><strong>Example</strong>:</p>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="nf">catcheR_filtercatch</span><span class="p">(</span>
<span class="w">    </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;docker&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;path/to/working/folder&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">expression.matrix</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;filename.csv&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">UMI.count</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">,</span>
<span class="w">    </span><span class="n">sample</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span>
<span class="p">)</span>
</pre></div>
</div>
<p><strong>Notes</strong>:</p>
<ul class="simple">
<li><p>This function reuses outputs from the initial <code class="docutils literal notranslate"><span class="pre">catcheR_10Xcatch</span></code> run.</p></li>
<li><p>Arguments and outputs are otherwise the same as <code class="xref py py-func docutils literal notranslate"><span class="pre">catcheR_10Xcatch()</span></code>.</p></li>
</ul>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">catcheR</a></h1>









<search id="searchbox" style="display: none" role="search">
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" placeholder="Search"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script><h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_design.html">catcheR_design - Oligonucleotide Design</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_step1QC.html">catcheR_step1QC - Pooled Cloning Step 1 QC</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_step2QC.html">catcheR_step2QC - Pooled Cloning Step 2 and hiPSC Genome Editing QC</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_scicounts.html">catcheR_scicounts - Obtain iPS2-sci-seq Count Matrices</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_10Xcatch.html">catcheR_10Xcatch - iPS2-10X-seq Perturbation Deconvolution</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_10XcatchQC.html">catcheR_10Xcatch - Fine-Tune iPS2-10X-seq Perturbation Assignment</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">catcheR_filtercatch - Fine-Tune iPS2-10X-seq Perturbation Assignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="catcheR_10Xnocatch.html">catcheR_10Xnocatch - Identify cells expressing no shRNA</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="catcheR_10XcatchQC.html" title="previous chapter">catcheR_10Xcatch - Fine-Tune iPS2-10X-seq Perturbation Assignment</a></li>
      <li>Next: <a href="catcheR_10Xnocatch.html" title="next chapter">catcheR_10Xnocatch - Identify cells expressing no shRNA</a></li>
  </ul></li>
</ul>
</div>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2025, Maria Luisa Ratto.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 8.1.3</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 1.0.0</a>
      
      |
      <a href="_sources/catcheR_filtercatch.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>